memory_engine/memory_store.py

```
@@ -114,7 +114,8 @@ def write(
         Raises:
             ValueError: If embedding is empty or metadata is not a dict
         """
-        if not embedding:
+        # Check if embedding is None or empty (handle numpy arrays properly)
+        if embedding is None or (hasattr(embedding, 'size') and embedding.size == 0):
             raise ValueError("Embedding cannot be empty")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)